src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplay.vb

```diff
@@ -115,30 +115,121 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+        ''' </remarks>
+        Public Function PrimitiveToMinimalDisplayParts(type As ITypeSymbol,
+                                                       value As Object,
+                                                       semanticModel As SemanticModel,
+                                                       position As Integer,
+                                                       Optional format As SymbolDisplayFormat = Nothing) As ImmutableArray(Of SymbolDisplayPart)
+            If Not (value Is Nothing OrElse value.GetType().IsPrimitive OrElse TypeOf value Is String OrElse TypeOf value Is Decimal OrElse TypeOf value Is Date) Then
+                Return Nothing
+            End If
+
+            Dim builder = ArrayBuilder(Of SymbolDisplayPart).GetInstance()
+            If type Is Nothing Then
+                AddConstantValue(builder, value, ToObjectDisplayOptions(If(format, SymbolDisplayFormat.VisualBasicErrorMessageFormat).ConstantValueOptions))
+            Else
+                Dim visitor = New SymbolDisplayVisitor(builder, If(format, SymbolDisplayFormat.VisualBasicErrorMessageFormat), semanticModel, position)
+                visitor.AddConstantValue(type, value)
+            End If
+
+            Return builder.ToImmutableAndFree()
+        End Function
+
+        Friend Sub AddConstantValue(builder As ArrayBuilder(Of SymbolDisplayPart), constantValue As Object, options As SymbolDisplayConstantValueOptions)
+            AddConstantValue(builder, constantValue, ToObjectDisplayOptions(options))
+        End Sub
+
+        Private Sub AddConstantValue(builder As ArrayBuilder(Of SymbolDisplayPart), constantValue As Object, options As ObjectDisplayOptions)
+            If constantValue IsNot Nothing Then
+                AddLiteralValue(builder, constantValue, options)
+            Else
+                builder.Add(New SymbolDisplayPart(SymbolDisplayPartKind.Keyword, Nothing, SyntaxFacts.GetText(SyntaxKind.NothingKeyword)))
+            End If
+        End Sub
+
+        Private Sub AddLiteralValue(builder As ArrayBuilder(Of SymbolDisplayPart), value As Object, options As ObjectDisplayOptions)
+            Debug.Assert(value.GetType().IsPrimitive OrElse value.GetType().IsEnum OrElse TypeOf value Is String OrElse TypeOf value Is Decimal OrElse TypeOf value Is Date)
+
+            Dim type = value.GetType()
+
+            Select Case type
+                Case GetType(String)
+                    AddSymbolDisplayParts(builder, DirectCast(value, String), options)
+
+                Case GetType(Char)
+                    AddSymbolDisplayParts(builder, DirectCast(value, Char), options)
+
+                Case Else
+                    Dim valueString = ObjectDisplay.FormatPrimitive(value, options)
+                    Debug.Assert(valueString IsNot Nothing)
+                    Dim kind = If(type = GetType(Boolean), SymbolDisplayPartKind.Keyword, SymbolDisplayPartKind.NumericLiteral)
```

Comment:
>type = GetType(Boolean) [](start = 34, length = 23)

VB has a ```TypeOf Is``` operation. Or perhaps use the same TypeCode value from previous checks.